demo/site/src/util/siteConfig.ts

```diff
@@ -40,3 +40,47 @@ export async function getSiteConfig() {
     if (!siteConfig) throw new Error(`SiteConfig not found for host ${host}`);
     return siteConfig;
 }
+
+type DomainInfo = {
+    domain: string;
+    previewData?: SitePreviewData;
+};
+
+export async function createEncodedDomain(headers: Headers) {
+    const host = getHostByHeaders(headers);
+    const sitePreviewParams = await previewParams({ skipDraftModeCheck: true });
+    const siteConfig = await getSiteConfigForHost(host);
+    if (!siteConfig) throw new Error(`SiteConfig not found for host ${host}`);
+
+    const domainInfo: DomainInfo = {
+        domain: siteConfig.scope.domain,
+        previewData: sitePreviewParams?.previewData,
+    };
+    return Buffer.from(JSON.stringify(domainInfo)).toString("base64");
+}
+
+type DecodedPageProps<T> = T & {
+    scope: ContentScope;
+    siteConfig: PublicSiteConfig;
+    previewData?: SitePreviewData;
+    isDraftModeEnabled: boolean;
+};
+
+export function decodePageProps<T extends { params: { domain: string; language?: string } }>(props: T): DecodedPageProps<T> {
```

Comment:
I'm not sure about "decoding" unrelated props. Maybe just decode the domain instead?